### PR TITLE
Feature/single tree contributions

### LINKED ISSUE #1813: Single Tree Contributions Page Redesign
### **Describe the feature you'd like to see included in Plant-for-the-Planet App.**

We need Single Tree Contributions Page to be updated with the following design - 

Screenshot 2020-01-15 at 12 50 58 PM

XD File is shared with @ankitecd , @sagararyal if you can upload this on Zeplin it would be easier. 

### **What are the use cases for this feature?**

**For clarity the different types of contributions are -** 

Trees which we donate via Projects
Trees which we donate via Gift Trees
Trees which we donate via Projects and are dedicated to someone 

Trees which are dedicated to us via Gift Trees
Trees which are dedicated to us via Dedicate Trees feature

Trees which are registered (Not donated) - Has edit and add measurement option
Trees which are registered (Not donated) and dedicated to someone - Has edit and add measurement option

moved the checkboxes to PR.
**@sagararyal are we removing NDVI from this page, currently it is present ?**

### PR DESCRIPTION
**Pull request by sagararyal**
Fixes #1813 #1155

### **What are various components and functions for this feature?**

# P1

- [x] Show date of the contribution

- [x] Show Image/Video Carousel - (Find the UI in Projects)

- [x] Show measurements (if available) only for contribution with Single Tree 

- [x] Give edit and delete function - If user has registered Trees

- [x] Show message for the various types of contributions listed above

- [x] Show Number of trees (Gifted/Received/Donated)

- [x] Show contact details of Plant Project - If user has not registered Trees (Find the UI in Projects)

- [x] Show Date of Donations

- [x] Show Tree Name, and Classification, if registered tree

- [x]  Hyperlink, project Names, and User Names, if gift, donations or dedications.

# P2

- [x] Show Map of the contribution

# P3
Comment the buttons and create a New issue for:
- [ ] Give option to download certificate - @sagararyal  what should the certificate contain ? 

- [ ] Give Share function - @sagararyal  what should the share message contain ? 

# P4
- [ ] Give add measurement feature

### CODE REVIEW

File: app/utils/utils.js [hunk 1]
```diff
@@ -65,6 +65,23 @@ export function formatDate(date, style = 'dd MMM yyyy', locale) {
   }
 }
 
+export function formatDateForContribution(
```
Comment (norbertschuler) [inline app/utils/utils.js]: We already have a library to format dates. Please use the same formatting to have a consistent UI for our users.

Comment (norbertschuler) [reply app/utils/utils.js]: I wonder why you don't use the above `formatDate` function with the style parameter given as `'MMMM d,  yyyy'`?

**Comment by ankitecd** [comment]
@sagararyal @jmiridis  

I need the following variables in the API of contribution to get the contact details of plant project done .

1. Slug of the plant project
2. url of the plant project
3. email of the plant project
4. address of the plant project

Screenshot 2020-01-27 at 10 43 29 AM

**Comment by sagararyal** [comment]
@ Harsh

I've got two changes and comments:

- [ ] Remove the buttons that says edit profile, and shows profile type.

Screen Shot 2020-01-29 at 11 42 32 AM

- [ ] Use the Gear icon, to show the side menu for now. (we will replace it in a separate change)

Are you planning to fix the save area view with pr on Top Bar?
Screen Shot 2020-01-29 at 11 33 30 AM

File: app/components/UserContributions/userContribution.native.js [hunk 2]
```diff
@@ -20,30 +27,129 @@ export default class UserContributions extends React.Component {
...
-        <View style={{ flex: 2, paddingTop: 20 }}>
+        <View style={styles.mapView}>
+          {this.getMapComponent(this.props.contribution)}
+          <TouchableOpacity
+            onPress={props.onClickClose}
+            style={[styles.button, styles.closeIcon]}
+          >
+            <View style={styles.closeContainer}>
+              <Image style={{ width: 16, height: 16 }} source={closeIcon} />
+            </View>
```
Comment (norbertschuler) [inline app/components/UserContributions/userContribution.native.js]: The design of the close icon should be the same UI for the whole app - therefore it should be aligned with what is used in #1792 taking this or the other solution, but do not build different solutions for every page.

File: app/components/UserContributions/userContribution.native.js [hunk 3]
```diff
@@ -20,30 +27,129 @@ export default class UserContributions extends React.Component {
...
+      latitudeDelta: 0.0000922,
+      longitudeDelta: 0.00421
+    };
+    let markerLatLong = {
+      latitude: geoLatitude,
+      longitude: geoLongitude
+    };
+    return (
+      <MapView
+        mapType={'satellite'}
```
Comment (norbertschuler) [inline app/components/UserContributions/userContribution.native.js]: Using the satellite map is causing more data traffic to the users loading the satellite images. @sagararyal do you want to do this here?

**Comment by norbertschuler** [comment]
- [ ] The Apple Maps Logo is partially covered (can be a reason for Apple to complain)
- [ ] The date does not get localized e.g. into German date format (display "January" instead of "Januar")
- [ ] The share link does not yet do anything

Simulator Screen Shot - iPhone 11 - 2020-01-30 at 15 46 46

File: app/components/UserContributions/userContribution.native.js (hunk 3 above)
Comment (sagararyal) [reply app/components/UserContributions/userContribution.native.js]: @norbertschuler the goal is to show the satellite images. we might allow an option to disable satellite in future.

File: app/utils/utils.js (hunk 1 above)
Comment (ankitecd) [reply app/utils/utils.js]: Sorry @norbertschuler my bad. I have removed the formatDateForContribution function and used formatDate with style param

**Pull request merged**